Ginger/GingerCoreCommon/UIElement/ElementInfo.cs

```diff
@@ -218,21 +219,83 @@ public string ElementTypeEnumDescription
             }
         }
 
+        List<String> mOptionalValues = new List<string>();
+        public List<String> OptionalValues
+        {
+            get
+            {
+                return mOptionalValues;
+            }
+            set
+            {
+                mOptionalValues = value;
+            }
+        }
+
+        ObservableList<OptionalValue> mOptionalVals = new ObservableList<OptionalValue>();
         [IsSerializedForLocalRepository]
-        public List<String> OptionalValues = new List<String>();
+        public ObservableList<OptionalValue> OptionalVals
+        {
+            get
+            {
+                if(mOptionalVals.Count == 0 && mOptionalValues.Count > 0)
+                {
+                    bool isDefault = false;
+                    foreach (string opVal in mOptionalValues)
+                    {
+                        mOptionalVals.Add(new OptionalValue() { ItemName = opVal, IsDefault = !isDefault });
+                        isDefault = true;
+                    }
+                    mOptionalValues = new List<string>();
+                }
+                return mOptionalVals;
+            }
+            set
+            {
+                mOptionalVals = value;
+            }
+        }
 
-        public string OptionalValuesAsString
+        public string OpValsString
         {
             get
             {
-                string listString = string.Empty;
-                foreach (string value in OptionalValues) listString += value + ",";
-                listString.TrimEnd(',');
-                return listString;
+                StringBuilder opValsString = new StringBuilder();
+                foreach (OptionalValue value in OptionalVals)
+                {
+                    if (value.IsDefault)
+                    {
+                        opValsString.Append(value.ItemName + "*,"); 
+                    }
+                    else
+                    {
+                        opValsString.Append(value.ItemName + ",");
+                    }
+                }
+                opValsString.ToString().TrimEnd(',');
+                return opValsString.ToString();
             }
         }
 
+        private void SetOptionalValuesFromOldObject()
+        {
```

Comment:
SetOptionalValuesFromOldObject() can be deleted. no?

Reply:
Removed